File: golang/lc00001-lc00100/lc00002/add_two_numbers_test.go
package lc00002

import "testing"

type ln ListNode

type caseType struct {
	l1       []int
	l2       []int
	expected int
}

func TestAddTwoNumbers(t *testing.T) {
	cases := []caseType{
		{[]int{0}, []int{0}, 0},
		{[]int{0}, []int{1}, 1},
		{[]int{1}, []int{0}, 1},
		{[]int{1, 0}, []int{1, 0, 2}, 112},
		{[]int{9, 4, 2}, []int{9, 4, 6, 5}, 10407},
	}

	for _, c := range cases {
		result := ln2Int(addTwoNumbers(int2Ln(c.l1), int2Ln(c.l2)))
		if result != c.expected {
			t.Fatalf("addTwoNumbers function failed, l1: %v, l2: %v, expected: %d, result: %d", c.l1, c.l2, c.expected, result)
		}
	}
}

func int2Ln(num []int) *ListNode {
	var tn *ListNode
	for i, length := 0, len(num); i < length; i++ {
		tn = &ListNode{num[i], tn}
	}
	return tn
}

func ln2Int(l *ListNode) int {
	num := 0
	for i := 1; l != nil; i *= 10 {
		num += l.Val * i
		l = l.Next
	}
	return num
}

File: php/lc00001-lc00100/lc00002/AddTwoNumbersTest.php
<?php

namespace leetcode\lc00002;

use PHPUnit\Framework\TestCase;

class AddTwoNumbersTest extends TestCase
{
    public function testAddTwoNumbers()
    {
        $cases = [
            ['l1' => $this->ln([9, 4, 2]), 'l2' => $this->ln([9, 4, 6, 5]), 'expected' => 10407],
            ['l1' => $this->ln([1, 0]), 'l2' => $this->ln([1, 0, 2]), 'expected' => 112],
            ['l1' => $this->ln([1]), 'l2' => $this->ln([0]), 'expected' => 1],
            ['l1' => $this->ln([0]), 'l2' => $this->ln([0]), 'expected' => 0],
        ];

        $obj = new AddTwoNumbers();
        for ($i = 0; $i < count($cases); $i++) {
            $this->assertEquals($cases[$i]['expected'], $this->ln2Int($obj->addTwoNumbers($cases[$i]['l1'], $cases[$i]['l2'])));
        }
    }

    public function ln($nums)
    {
        $ln = null;
        for ($i = 0; $i < count($nums); $i++) {
            $ln = new ListNode($nums[$i], $ln);
        }
        return $ln;
    }

    public function ln2Int($ln)
    {
        $num = 0;
        $i = 1;
        while ($ln !== null) {
            $num += ($ln->val * $i);
            $ln = $ln->next;
            $i *= 10;
        }
        return $num;
    }
}
